cmd/kubeadm/app/cmd/join.go

```diff
@@ -433,14 +433,9 @@ func (j *Join) CheckIfReadyForAdditionalControlPlane(initConfiguration *kubeadma
 		return errors.New("unable to add a new control plane instance a cluster that doesn't have a stable controlPlaneEndpoint address")
 	}
 
-	// blocks if control plane is self-hosted
-	if features.Enabled(initConfiguration.FeatureGates, features.SelfHosting) {
-		return errors.New("self-hosted clusters are deprecated and won't be supported by `kubeadm join --experimental-control-plane`")
-	}
-
-	// blocks if the certificates for the control plane are stored in secrets (instead of the local pki folder)
-	if features.Enabled(initConfiguration.FeatureGates, features.StoreCertsInSecrets) {
-		return errors.New("certificates stored in secrets, as well as self-hosted clusters are deprecated and won't be supported by `kubeadm join --experimental-control-plane`")
+	// blocks if the cluster was created without an external etcd cluster
+	if initConfiguration.Etcd.External == nil {
```

Comment:
This was added during a rebase. it is not something I wrote and was honestly confused by why git was doing what it was doing. I had to commit it to see how it would show up on the PR. (thought it would not show me as adding it.) anyhow. will update this as you are correct it is not part of the PR.

Reply:
I think you are correct. I dug thought the git history and believe that is what happened. it should be corrected now.